perllib/FixMyStreet/Roles/DDProcessor.pm

```diff
@@ -117,21 +117,19 @@ sub waste_reconcile_direct_debits {
                 next unless $self->waste_is_dd_payment($cur);
                 # only confirmed records are valid.
                 next unless FixMyStreet::DB::Result::Problem->visible_states()->{$cur->state};
+                # already processed
+                next RECORD if $cur->get_extra_metadata('dd_date') && $cur->get_extra_metadata('dd_date') eq $payment->date;
+                next if $p;
```

Comment:
I guess the only reason to not do that is if we want it to log all the potentially matching reports.

Reply:
I did have a last here (or actually, at the bottom of the loop when $p got set), but I took it out because otherwise it won't skip to the `next RECORD` should it ever find the matching dd_date one *after* finding the $p, iyswim (which I think it did in a test, though seems like that wouldn't happen in real life).

Reply:
Ah yes. As you say it's not clear that ever happens but I guess it's a not unreasonable safety check.